utils/self-extracting-executable/compressor.cpp

```diff
@@ -9,7 +9,18 @@
 #include <cerrno>
 #include <memory>
 #include <iostream>
+#if defined __APPLE__
```

Comment:
ClickHouse uses it's own platform macros which are also available in the self-extracting executable (I tried it out).

```suggestion
#if defined OS_DARWIN
```